src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultValidationHtmlAttributeProvider.cs

```diff
@@ -14,7 +13,7 @@ namespace Microsoft.AspNetCore.Mvc.ViewFeatures
     /// <summary>
     /// Default implementation of <see cref="ValidationHtmlAttributeProvider"/>.
     /// </summary>
-    public class DefaultValidationHtmlAttributeProvider : ValidationHtmlAttributeProvider
+    internal sealed class DefaultValidationHtmlAttributeProvider : ValidationHtmlAttributeProvider
```

Comment:
Suggestions?

Reply:
I suggest this should be `public` since it's difficult to extend the validation system without it. If that means we expose `ClientValidatorCache` or a new `IClientValidatorCache` interface, fine.